params/config.go

```diff
@@ -567,9 +567,9 @@ func (c *ChainConfig) IsTerminalPoWBlock(parentTotalDiff *big.Int, totalDiff *bi
 	return parentTotalDiff.Cmp(c.TerminalTotalDifficulty) < 0 && totalDiff.Cmp(c.TerminalTotalDifficulty) >= 0
 }
 
-// IsShanghai returns whether num is either equal to the Shanghai fork block or greater.
-func (c *ChainConfig) IsShanghai(num *big.Int) bool {
-	return isForked(c.ShanghaiBlock, num)
+// IsShanghai returns whether time is either equal to the Shanghai fork time or greater.
+func (c *ChainConfig) IsShanghai(time *big.Int) bool {
```

Comment:
```suggestion
func (c *ChainConfig) IsShanghai(time uint64) bool {
```
It would be nice if either this took a `uint64` or if we don't want `IsShanghai` to take a different parameter  type than other `isForked` methods, maybe a new getter on `types.Block` and `types.Header` "`TimeBig()`" so that each time we check `IsShanghai`, don't need to jump through a bunch of casting hoops?

Reply:
It might be better to have a different type here to detect at compile time if someone is trying to pass in a block number instead of a timestamp.  e.g. I see at least one such call in core/chain_makers.go that should be updated.